torch/csrc/autograd/engine.cpp

```diff
@@ -871,7 +871,12 @@ auto Engine::execute(const edge_list& roots,
     graph_task->init_to_execute(*graph_root, outputs);
   }
 
-  return execute_with_graph_task(graph_task, graph_root)->wait();
+  execute_with_graph_task(graph_task, graph_root);
+  // Avoid a refcount bump for the Future, since we check for refcount in
```

Comment:
I'm not sure what this comment refers to? Why does it talk about the distributed engine?

Reply:
It is referring to this check in DistEngine: https://github.com/pytorch/pytorch/blob/master/torch/csrc/distributed/autograd/engine/dist_engine.cpp#L559. If we incur a refcount bump here, that check would fail.

Reply:
Maybe refer to that line or a PR talking about this because I don't think it is clear otherwise :/

Reply:
I updated the PR to include some more details about this check in DistEngine referring to the line we're talking about here.